### PR TITLE
Uniset contains bugfix

### LINKED ISSUE #268: [uniset] Crash when testing for non-first range
```rust
use icu_unicodeset::UnicodeSet;
use icu_unicodeset::UnicodeSetBuilder;
use std::char;

fn main() {
    let ranges = vec!['A'..='Z', 'a'..='z'];
    let mut builder = UnicodeSetBuilder::new();

    for range in &ranges {
        builder.add_range(range);
    }

    let set = builder.build();

    assert_eq!(set.contains_range(&('c'..='e')), true);
}
```

crashes with:

```
thread 'main' panicked at 'index out of bounds: the len is 4 but the index is 4', /Users/zbraniecki/projects/icu4x/components/uniset/src/uniset.rs:189:36
```

full stack:

```
thread 'main' panicked at 'index out of bounds: the len is 4 but the index is 4', /Users/zbraniecki/projects/icu4x/components/uniset/src/uniset.rs:189:36
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::panicking::panic_bounds_check
   9: <usize as core::slice::SliceIndex<[T]>>::index
  10: core::slice::<impl core::ops::index::Index<I> for [T]>::index
  11: <alloc::vec::Vec<T> as core::ops::index::Index<I>>::index
  12: icu_unicodeset::uniset::UnicodeSet::contains_range
  13: uniset_test::main
  14: std::rt::lang_start::{{closure}}
  15: std::rt::lang_start_internal
  16: std::rt::lang_start
  17: main
```

### PR DESCRIPTION
**Pull request by EvanJP**
Fixes #268 

TLDR: Off by one error

The documentation for `contains_query` is that the value in the returned `Option` will be the _start_ index of the range that contains the query. 

When performing `std::Vec::Vec::binary_search(&T)`, the index returned is the upper bound index of where the query should be:

E.g.
```rust
let test = vec![0, 10, 15, 20];
assert_eq!(test.binary_search(&1), Err(1))
```

In the implementation of `contains_query`, the issue of dropping the index back to the lower bound of the range was ignored, causing #268 .

Before this PR:
```rust
let check = UnicodeSet::from_inversion_list(vec![65, 70, 75, 80]);
assert_eq!(check.contains_query(66), Some(1)); // INCORRECT BEHAVIOR
```

After after PR:
```rust
let check = UnicodeSet::from_inversion_list(vec![65, 70, 75, 80]);
assert_eq!(check.contains_query(66), Some(0)); // INTENDED BEHAVIOR
```

### CODE REVIEW

**Review by zbraniecki — DISMISSED** [review]
lgtm!

File: components/uniset/src/uniset.rs [hunk 1]
```diff
@@ -331,7 +343,9 @@ mod tests {
         let ex = vec![65, 70, 75, 85];
         let check = UnicodeSet::from_inversion_list(ex).unwrap();
         assert!(check.contains_range(&('A'..='E'))); // 65 - 69
-        assert!(check.contains_range(&('K'..'U'))); // 75 - 84
+        assert!(check.contains_range(&('C'..'D'))); // 65 - 69
```
Comment (echeran) [inline components/uniset/src/uniset.rs:346]: Fix code point numbers in comment.

**Review by echeran — APPROVED** [review]
Looks good, but fix source comment first.

File: components/uniset/src/uniset.rs (hunk 1 above)
Comment (EvanJP) [reply components/uniset/src/uniset.rs]: good catch, agreed and changed

**Review by echeran — APPROVED** [review]
